### PR TITLE
Add dockcross support for linux-{x86,x64} and windows-{x86,x64}

### PR DESCRIPTION
**Pull request by brainstorm**
Cleanup docker releng infrastructure by using dockcross. This is accompanied with another pull request on radare/radare2:

https://github.com/radare/radare2/pull/6342

### CODE REVIEW

File: dockcross [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    docker pull $FINAL_IMAGE
+}
+
+help:update-image() {
+    echo Pull the latest $FINAL_IMAGE .
+}
+
+command:update-script() {
+    if cmp -s <( docker run $FINAL_IMAGE ) $0; then
+        echo $0 is up to date
```
Comment (radare) [inline dockcross]: quote the argument of echo

File: dockcross [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    else
+        echo -n Updating $0 '... '
```
Comment (radare) [inline dockcross]: use printf instead of echo -n. be posix my friend

File: dockcross [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+command:update-image() {
+    docker pull $FINAL_IMAGE
+}
+
+help:update-image() {
+    echo Pull the latest $FINAL_IMAGE .
+}
+
+command:update-script() {
+    if cmp -s <( docker run $FINAL_IMAGE ) $0; then
```
Comment (radare) [inline dockcross]: isnt $1 what you wanna use here?

File: dockcross [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        echo $0 is up to date
+    else
+        echo -n Updating $0 '... '
+        docker run $FINAL_IMAGE > $0 && echo ok
+    fi
+}
+
+help:update-image() {
+    echo Update $0 from $FINAL_IMAGE .
```
Comment (radare) [inline dockcross]: use tabs

File: build.sh [hunk 5]
```diff
@@ -81,16 +81,30 @@ linux_build() {(
...
 )}
 
 docker_linux_build() {(
 	arch="$1"
 	check radare2_${VERSION}_${arch}.deb && return
-	cd docker-linux-${arch} || exit 1
-	make
+	cd tmp/radare*
+#	${CWD}/dockcross --image dockcross/linux-${arch} ./configure --with-compiler=${arch} --host=${arch}
+	${CWD}/dockcross --image dockcross/linux-${arch} sys/build.sh
```
Comment (radare) [inline build.sh]: CWD can contain spaces

File: dockcross [hunk 6]
```diff
@@ -0,0 +1,234 @@
+#!/bin/bash
+
+DEFAULT_DOCKCROSS_IMAGE=dockcross/linux-armv5
+
+#------------------------------------------------------------------------------
+# Helpers
+#
+err() {
+    echo -e >&2 ERROR: $@\\n
```
Comment (radare) [inline dockcross]: echo -e is not portable

File: dockcross [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+die() {
+    err $@
+    exit 1
+}
+
+has() {
+    # eg. has command update
+    local kind=$1
```
Comment (radare) [inline dockcross]: local is a bashism

File: dockcross [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    local name=$2
+
+    type -t $kind:$name | grep -q function
```
Comment (radare) [inline dockcross]: type -t is another bashism what you wanna do here?

File: dockcross [hunk 9]
```diff
@@ -0,0 +1,234 @@
+#!/bin/bash
```
Comment (radare) [inline dockcross]: use /bin/sh plz bash is cancer

File: dockcross (hunk 3 above)
Comment (brainstorm) [inline dockcross]: Don't think so... it's looking wether the script `dockcross` is up to date, isn't it?

File: dockcross (hunk 4 above)
Comment (brainstorm) [inline dockcross]: where? in the whole script? between which arguments? context plz :-S

File: build.sh (hunk 5 above)
Comment (brainstorm) [inline build.sh]: Is there a cleaner way to trim those other than using ugly `${text// /}` hacks, sed or similar?

**Comment by radare** [comment]
Can you rebase on top of master?

File: dockcross [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+            ;;
+
+    esac
+fi
+
+# Set the docker run extra args (if any)
+FINAL_ARGS=${ARG_ARGS-${DOCKCROSS_ARGS}}
+
+# Bash on Ubuntu on Windows
+UBUNTU_ON_WINDOWS=$([ -e /proc/version ] && grep -l Microsoft /proc/version || echo "")
```
Comment (radare) [inline dockcross]: lol

**Comment by radare** [comment]
Oh wow! can you try enabling https://github.com/dockcross/dockcross/tree/master/browser-asmjs ? :D

**Pull request merged**